crm_construction_site/models.py

```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+
+from openerp import models, fields, api
+
+
+class Lead(models.Model):
+    _name = 'crm.lead'
+    _inherit = 'crm.lead'
+
+    site = fields.Many2one('res.partner')
+
+    @api.onchange('partner_id')
```

Comment:
I tried that, didn't work.
I think it was conflicting with on_change_partner_id() function, which was directly called from the view. I have overriden that function now.